src/TextFieldRich.jsx

```
@@ -1,236 +1,455 @@
+		onKeyDown: React.PropTypes.func,
+		onEnterKeyDown: React.PropTypes.func,
+		autoGrow: React.PropTypes.bool,
+		height: React.PropTypes.number,
+		value: React.PropTypes.string,
+	},
+
+	getDefaultProps: function() {
+		return {
+			autoGrow: true,
+			value: "Enter description here.",
+		};
+	},
+	
+	getInitialState: function() {
+    	return {
+    		displayCodeMirror: false,
+    	};
+    },
+
+	componentDidMount: function() {
+		var iframe = this.refs.rte.getDOMNode();
+		iframe.style.width = "100%"; // temporary
+		
+		// If height has been passed then set the iframe height
+		if (this.props.height) {
+			iframe.style.height = height;
+		}
+				
+		// Make the contents of the iframe editable
+		var cls = this;
+		setTimeout(function() { 
+					cls._enableDesign(true);
+					cls._setValue(cls.props.value);
+				}, 1);
+		
+		//this._setupIframeDocument(true); // this has issue when executing the function instantly 
+	},
+	
+	componentDidUpdate: function() {
+		
+		// Re enable the content editable of the iframe document
+		var cls = this;
+		setTimeout(function() {
+					cls._enableDesign(true);
+					cls._setValue(cls._currentValue);
+				}, 1);
+		
+	},
+
+	render: function() {
+		
+		var display = null;
+		
+		// Determine what should be displayed
+		if(this.state.displayCodeMirror) {
+			display = <CodeMirror ref="codemirror" options={{lineNumbers: true,  mode: "text/html"}} />;
+		}
+		else {
+			display = <iframe ref="rte" src="about:blank" />;
+		}
+		
+		return ( 
+				<div className="chamel-text-field-rich" >{display}</div>
+		);
+	},
+	
+	/**
+	 * Public interface to send commands to the RTE
+	 *
+	 * @param {string} command The name of the RTE command to execute
+	 * @public
+	 */
+	_sendCommand: function (command, option) {
+		try {
+			var idoc = this._getIframeDoc();
+			var iwnd = this._getIframeWindow();
+		
+			iwnd.focus();
+		  	idoc.execCommand(command, false, option);
+			iwnd.focus();
+		} 
+		catch (e) {
+			alert(e);
```

This will be fired any time _sendCommand is called if in codeMirror mode. Maybe rather than show an alert, we should throw an exception if we don't expect anyone to ever call this in codeMirrorMode.